springwolf-plugins/springwolf-kafka-plugin/src/main/java/io/github/springwolf/plugins/kafka/asyncapi/scanners/bindings/KafkaBindingFactory.java

```diff
@@ -36,4 +57,80 @@ public Map<String, OperationBinding> buildOperationBinding(KafkaListener annotat
     public Map<String, MessageBinding> buildMessageBinding(KafkaListener annotation, SchemaObject headerSchema) {
         return KafkaListenerUtil.buildMessageBinding(headerSchema);
     }
+
+    @Override
+    public void setApplicationContext(@Nonnull ApplicationContext applicationContext) throws BeansException {
```

Comment:
To keep this class simple, can all of this code be pushed down into an own class, like KafkaBeanRefHelper, etc (I am open for better naming)

This allows to provide a link where spring-kafka implements beanRef, while creating an own, lightweight implementation.
Additional the class can implement the stringValueResolver interface, so that it can be used in the original way.